File: NeuroLearnX/server/routes/authRoutes.js
/**
 * Auth API routes (prefix: /api/auth).
 */
const express = require("express");
const {
  registerStudent,
  login,
  loginStudent,
  loginAdmin,
  logout,
  me,
} = require("../controllers/authController");
const { requireAuth } = require("../middleware/authMiddleware");

const router = express.Router();

router.post("/student/register", registerStudent);
router.post("/login", login);
router.post("/student/login", loginStudent);
router.post("/admin/login", loginAdmin);
router.post("/logout", logout);
router.get("/me", requireAuth, me);

module.exports = router;

File: NeuroLearnX/nlx/routes/auth_routes.py
from flask import Blueprint

from ..controllers import auth_controller

auth_bp = Blueprint("auth_bp", __name__, url_prefix="/api/auth")

# ── Registration ──────────────────────────────────────────────────────────
auth_bp.post("/student/register")(auth_controller.register_student)
auth_bp.post("/register")(auth_controller.register)

# ── Login ─────────────────────────────────────────────────────────────────
auth_bp.post("/login")(auth_controller.login)           # unified (any role)
auth_bp.post("/admin/login")(auth_controller.login_admin)    # admin only
auth_bp.post("/student/login")(auth_controller.login_student) # student only

# ── Session ───────────────────────────────────────────────────────────────
auth_bp.post("/logout")(auth_controller.logout)
auth_bp.get("/me")(auth_controller.me)

# ── Admin-only test ───────────────────────────────────────────────────────
auth_bp.get("/admin/ping")(auth_controller.admin_only_ping)
